Strange compilation error while migrating to RTIC v1

I've tried to migrate to rtic v1 here: https://github.com/TeXitoi/keyseebee/tree/rtic1 but I have a strange error:

```
$ cargo build --release 
   Compiling keyseebee v0.1.0 (.../keyseebee/firmware)
error[E0080]: evaluation of constant value failed
  --> src/main.rs:42:1
   |
42 | #[app(device = crate::hal::pac, peripherals = true, dispatchers = [CEC_CAN])]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `4_usize - 5_usize`, which would overflow
   |
   = note: this error originates in the attribute macro `app` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not compile `keyseebee` due to previous error
```

I've tried to force using the v1.0.0 version, but I have the same error:

```
$ git diff Cargo.toml
diff --git a/firmware/Cargo.toml b/firmware/Cargo.toml
index 8026e51..c3b6091 100644
--- a/firmware/Cargo.toml
+++ b/firmware/Cargo.toml
@@ -10,11 +10,15 @@ cortex-m = "0.7"
 cortex-m-rt = { version = "0.7.1", features = ["device"] }
 panic-halt = "0.2.0"
 keyberon = { git = "https://github.com/TeXitoi/keyberon", branch = "master" }
-cortex-m-rtic = "1.1.1"
+cortex-m-rtic = "=1.0.0"
+cortex-m-rtic-macros = "=1.0.0"
 embedded-hal = "0.2"
 usb-device = "0.2.0"
 nb = "1.0.0"
 
+[dev-dependencies]
+rtic-syntax = "=1.0.0"
+
 [profile.release]
 lto = true
 incremental = false
$ cargo outdated
    Updating git repository `https://github.com/TeXitoi/keyberon`
Name                                 Project  Compat  Latest  Kind         Platform
----                                 -------  ------  ------  ----         --------
cortex-m-rtic                        1.0.0    ---     1.1.1   Normal       ---
cortex-m-rtic->cortex-m-rtic-macros  1.0.0    ---     1.1.0   Normal       ---
cortex-m-rtic-macros                 1.0.0    ---     1.1.0   Normal       ---
cortex-m-rtic-macros->rtic-syntax    1.0.0    ---     1.0.1   Normal       ---
rtic-syntax                          1.0.0    ---     1.0.1   Development  ---
```

--- a/macros/src/codegen/pre_init.rs
+++ b/macros/src/codegen/pre_init.rs
@@ -49,8 +49,14 @@ pub fn codegen(app: &App, analysis: &Analysis, extra: &Extra) -> Vec<TokenStream
             Some((&task.args.priority, &task.args.binds))
         }
     })) {
+        let es = format!(
+            "Maximum priority used by interrupt vector '{}' is more than supported by hardware",
+            name
+        );
         // Compile time assert that this priority is supported by the device
-        stmts.push(quote!(let _ = [(); ((1 << #nvic_prio_bits) - #priority as usize)];));
+        stmts.push(quote!(
+            const _: () = assert!((1 << #nvic_prio_bits) >= #priority as usize, #es);
+        ));
 
         stmts.push(quote!(
             core.NVIC.set_priority(
@@ -72,8 +78,14 @@ pub fn codegen(app: &App, analysis: &Analysis, extra: &Extra) -> Vec<TokenStream
             None
         }
     }) {
+        let es = format!(
+            "Maximum priority used by interrupt vector '{}' is more than supported by hardware",
+            name
+        );
         // Compile time assert that this priority is supported by the device
-        stmts.push(quote!(let _ = [(); ((1 << #nvic_prio_bits) - #priority as usize)];));
+        stmts.push(quote!(
+            const _: () = assert!((1 << #nvic_prio_bits) >= #priority as usize, #es);
+        ));
 
         stmts.push(quote!(core.SCB.set_priority(
             rtic::export::SystemHandler::#name,
@@ -90,8 +102,15 @@ pub fn codegen(app: &App, analysis: &Analysis, extra: &Extra) -> Vec<TokenStream
         };
         let binds = &monotonic.args.binds;
 
+        let name = &monotonic.ident;
+        let es = format!(
+            "Maximum priority used by monotonic '{}' is more than supported by hardware",
+            name
+        );
         // Compile time assert that this priority is supported by the device
-        stmts.push(quote!(let _ = [(); ((1 << #nvic_prio_bits) - #priority as usize)];));
+        stmts.push(quote!(
+            const _: () = assert!((1 << #nvic_prio_bits) >= #priority as usize, #es);
+        ));
 
         let mono_type = &monotonic.ty;
 

--- a/ui/task-priority-too-high.stderr
+++ b/ui/task-priority-too-high.stderr
@@ -1,7 +1,15 @@
+warning: unused variable: `cx`
+  --> ui/task-priority-too-high.rs:12:13
+   |
+12 |     fn init(cx: init::Context) -> (Shared, Local, init::Monotonics) {
+   |             ^^ help: if this is intentional, prefix it with an underscore: `_cx`
+   |
+   = note: `#[warn(unused_variables)]` on by default
+
 error[E0080]: evaluation of constant value failed
- --> $DIR/task-priority-too-high.rs:3:1
+ --> ui/task-priority-too-high.rs:3:1
   |
 3 | #[rtic::app(device = lm3s6965)]
-  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `8_usize - 9_usize`, which would overflow
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Maximum priority used by interrupt vector 'I2C0' is more than supported by hardware', $DIR/ui/task-priority-too-high.rs:3:1
   |
-  = note: this error originates in the attribute macro `rtic::app` (in Nightly builds, run with -Z macro-backtrace for more info)
+  = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)

